python/deltalake/table.py

```diff
@@ -456,6 +456,27 @@ def vacuum(
             enforce_retention_duration,
         )
 
+    def update(
+        self,
+        updates: Dict[str, str],
+        predicate: Optional[str] = None,
+        writer_properties: Optional[Dict[str, int]] = None,
```

Comment:
Once we [drop support for Python 3.7](https://github.com/delta-io/delta-rs/issues/1703), I'd love for this to be a [TypedDict](https://peps.python.org/pep-0589/)

Reply:
I'll make a note of it, let me know if there is a PR that is dropping the support so I can track it.